tests/system_tests/test_core/test_wandb_metadata.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+import wandb
+
+
+@pytest.mark.parametrize("disabled", [False, True])
+def test_metadata_ops(user, disabled: bool):
+    run = wandb.init(
+        settings=wandb.Settings(
+            mode="disabled" if disabled else "online",
+            x_stats_sampling_interval=1,
+        ),
+    )
+
+    if run.settings.x_require_legacy_service:
+        # Legacy service does not support metadata
+        assert run._metadata is None
+        run.finish()
+        return
```

Comment:
maybe just split into 2 tests and use markers? i think it will be more readable?